### PR TITLE
7/save actions

### PR DESCRIPTION
**Pull request by pyphilia**
Reopen PR due to branch renaming.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+// identify and check the correct endpoint of the request
+export const createAction = async function (request, reply, runner, itemTaskManager) {
+  // check that the request is ok
+  if (correctStatusCodes.includes(reply.statusCode)) {
+    const url: string = request.url;
+    const method: string = request.method;
+    const member: Member = request.member;
+    const paramItemId: string = request.params.id;
+    const queryItemsId: string[] = request.query.id;
+    const geo = geoip.lookup(request.ip);
```
Comment (juancarlosfarah) [inline src/index.ts:182]: We need to somehow allow this to be disabled.

File: src/services/action/base-action-task.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  input: unknown;
+  getInput: () => unknown;
+
+  getResult: () => unknown;
+
+  constructor(actor: Actor, actionService: ActionService) {
+    this.actor = actor;
+    this.actionService = actionService;
+    this.status = 'NEW';
```
Comment (juancarlosfarah) [inline src/services/action/base-action-task.ts:27]: Factor this out?

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Interesting commit! Did my best to try and follow everything that's happening. Added questions, comments, and suggestions.

File: src/services/action/get-actions-task.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      // Get actions with requestedSampleSize and view
+      actionResult = await this.actionService.getActionsByItemWithSample(
+        this.itemId,
+        sampleSize,
+        handler,
+      );
+    }
+
+    this._result = actionResult;
+    this.status = 'OK';
```
[the remark this answers is not in the document]
Comment (pyphilia) [reply src/services/action/get-actions-task.ts:75]: Kind of, but it's only in graasp core. We plan to have a graasp-constants.. or in graasp-types to uniformize everything.

File: src/db-service.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    requestedSampleSize: number,
+    transactionHandler: TrxHandler,
+  ): Promise<readonly Action[]> {
+    return transactionHandler
+      .query<Action>(
+        sql`
+        SELECT ${ActionService.allColumns} 
+        FROM action
+        WHERE "item_id" = ${itemId}
+        ORDER BY RANDOM()
```
[the remark this answers is not in the document]
Comment (pyphilia) [reply src/db-service.ts:108]: It's in case there's more actions than the sample we want to take. Wasn't it random in the old graasp?

**Pull request merged**